cachi2/core/package_managers/yarn/main.py

```diff
@@ -73,21 +75,25 @@ def _configure_yarn_version(project: Project) -> None:
             ),
         )
 
-    if (
-        yarn_path_version
-        and package_manager_version
-        and yarn_path_version != package_manager_version
-    ):
-        raise PackageRejected(
-            (
-                f"Mismatch between the yarn versions specified by yarnPath (yarn@{yarn_path_version}) "
-                f"and packageManager (yarn@{package_manager_version})"
-            ),
-            solution=(
-                "Ensure that the versions of yarn specified by yarnPath in .yarnrc.yml and "
-                "packageManager in package.json agree"
-            ),
-        )
+    if yarn_path_version and package_manager_version:
+        if yarn_path_version != package_manager_version:
+            raise PackageRejected(
+                (
+                    f"Mismatch between the yarn versions specified by yarnPath (yarn@{yarn_path_version}) "
+                    f"and packageManager (yarn@{package_manager_version})"
+                ),
+                solution=(
+                    "Ensure that the versions of yarn specified by yarnPath in .yarnrc.yml and "
+                    "packageManager in package.json agree"
+                ),
+            )
+    else:
+        version = yarn_path_version if yarn_path_version else package_manager_version
+        if version < VersionInfo(3, 0, 0) or version >= VersionInfo(4, 0, 0):  # type: ignore # version cannot be None because we already checked both operands above
```

Comment:
A way to avoid completely ignoring the type here is to add a dummy conditional such as:
```
if not version:
   raise RuntimeError("")
```

In case you prefer to keep the `type: ignore` solution, can you move the comment explaining why it is needed to a different line? This line currently is much larger than our usual 100 characters limit (not sure if there's a way to configure our linters to catch that though).

Reply:
One last suggestion is to narrow down what we're ignoring to only `# type: ignore[operator]`

Reply:
>```
> if not version:
>   raise RuntimeError("")
>```

We shouldn't add dead code just because mypy isn't clever enough to figure out version cannot be None.